bluepyemodel/evaluation/efel_feature_bpem.py

```diff
@@ -223,8 +224,253 @@ def calculate_score(self, responses, trace_check=False):
+        decay=None,
+        linear=None,
+    ):
+        """Constructor"""
+        # pylint: disable=too-many-arguments
+        super().__init__(
+            name,
+            efel_feature_name,
+            recording_names,
+            stim_start,
+            stim_end,
+            exp_mean,
+            exp_std,
+            threshold,
+            stimulus_current,
+            comment,
+            interp_step,
+            double_settings,
+            int_settings,
+            string_settings,
+        )
+        self.decay = decay
+        self.linear = linear
+        self.ymult = None  # set on the fly before the fitting
+
+    @property
+    def recording_names_list(self):
+        return self.recording_names.values()
+
+    @property
+    def distances(self):
+        # expects keys in recordings names to be distances from soma (e.g. "50") or "" if at soma
+        return [int(rec_name) if rec_name != "" else 0 for rec_name in self.recording_names.keys()]
+
+    @property
+    def locations(self):
+        # locations used in bpo features (holding current, threshold current)
+        # as such, they are implemented for MultiProtocols, but not for simple protocol
+        raise NotImplementedError
+
+    def _construct_efel_trace(self, responses):
+        """Construct trace that can be passed to eFEL"""
+
+        traces = []
+        if "" not in self.recording_names:
+            raise ValueError("eFELFeature: '' needs to be in recording_names")
+
+        for recording_name in self.recording_names_list:
+            if recording_name not in responses:
+                logger.debug(
+                    "Recording named %s not found in responses %s", recording_name, str(responses)
+                )
+                return None
+
+            if responses[recording_name] is None:
+                logger.debug("resp of %s is None", responses[recording_name])
+                return None
+
+            trace = {}
+            trace["T"] = responses[recording_name]["time"]
+            trace["V"] = responses[recording_name]["voltage"]
+            if callable(self.stim_start):
+                trace["stim_start"] = [self.stim_start()]
+            else:
+                trace["stim_start"] = [self.stim_start]
+
+            if callable(self.stim_end):
+                trace["stim_end"] = [self.stim_end()]
+            else:
+                trace["stim_end"] = [self.stim_end]
+            traces.append(trace)
+
+        return traces
+
+    def exp_decay(self, x, p):
+        return numpy.exp(-x / p) * self.ymult
+
+    def exp(self, x, p):
+        return numpy.exp(x / p) * self.ymult
+
+    def linear_fit(self, x, p):
+        return self.ymult + p * x
+
+    def fit(self, distances, values):
+        """Fit back propagation"""
+        guess = [50]
+        self.ymult = values[distances.index(0)]
+        if self.linear:
+            params, _ = opt.minpack.curve_fit(self.linear_fit, distances, values, p0=guess)
```

Comment:
Here you use scipy.optimize.minpack.curve_fit and not scipy.optimize.curve_fit. is there any special reason? I haven't used the one you use

Reply:
good catch! I'll use opt.curve_fit starting next commit